### PR TITLE
Add RSA signature

### PR DESCRIPTION
**Pull request by lucklove**
Signed-off-by: lucklove <gnu.crazier@gmail.com>

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### CODE REVIEW

File: pkg/repository/crypto/keys.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	// ErrorKeyUninitialized will be present when key is used before Deserialize called
+	ErrorKeyUninitialized = errors.New("Key not initialized, call Deserialize first")
+	// ErrorDeserializeKey means the key format is not valid
+	ErrorDeserializeKey = errors.New("Error on deserialize key, check if the key is valid")
+)
+
+// KeyStore is the collection of all public keys available to TiUp.
+type KeyStore interface {
+	// Add put new key into KeyStore
+	Add(string, PubKey) KeyStore
```
Comment (lonng) [inline pkg/repository/crypto/keys.go:30]: Prefer `Put`

File: pkg/repository/crypto/keys.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+// PubKey is a public key available to TiUp
+type PubKey interface {
+	Serializable
+	// Verify check the signature is right
+	Verify(payload []byte, sig string) error
+}
+
+// PriKey is the private key that provide signature method
+type PriKey interface {
```
Comment (AstroProfundis) [inline pkg/repository/crypto/keys.go:53]: I think `PrivKey` might be better.

File: pkg/repository/crypto/rsa.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/asn1"
+	"encoding/base64"
+	"encoding/pem"
+)
+
+// RsaPair generate a pair of rsa keys
+func RsaPair() (*RsaPubKey, *RsaPriKey, error) {
```
Comment (AstroProfundis) [inline pkg/repository/crypto/rsa.go:27]: Should be `RSA` with all tree captive letters

File: pkg/repository/crypto/rsa.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	key, err := rsa.GenerateKey(rand.Reader, 2048)
```
Comment (AstroProfundis) [inline pkg/repository/crypto/rsa.go:28]: Put the key length to a const at top? We may need to change the value (or even support different values) in the future

File: pkg/repository/crypto/keys.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+type Serializable interface {
+	// Translate the key to the format that can be stored
+	Serialize() ([]byte, error)
+
+	// Deserialize a key from data
+	Deserialize([]byte) error
+}
+
+// PubKey is a public key available to TiUp
+type PubKey interface {
```
Comment (AstroProfundis) [inline pkg/repository/crypto/keys.go:46]: Should also add a method to get fingerprint of the key

File: pkg/sign/sign.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package sign
```
Comment (lonng) [inline pkg/sign/sign.go:1]: Please add the license header.

File: pkg/repository/crypto/keys.go (hunk 5 above)
Comment (lucklove) [reply pkg/repository/crypto/keys.go:46]: Why need this now?

File: pkg/sign/sign.go (hunk 6 above)
Comment (lucklove) [reply pkg/sign/sign.go]: This file is of no use, deleted already

File: pkg/repository/crypto/keys.go (hunk 5 above)
Comment (lucklove) [reply pkg/repository/crypto/keys.go:46]: The key-id has the same function as this, but it should be added into KeyInfo, I think

Comment (AstroProfundis) [reply pkg/repository/crypto/keys.go:46]: I believe the value of key id is fingerprint of that key, I didn't see that part, maybe I missed something?

**Pull request merged**